app/kontres/models/reservation.py

```diff
@@ -0,0 +1,67 @@
+import uuid
+
+from django.db import models
+
+from app.common.enums import Groups
+from app.common.permissions import BasePermissionModel
+from app.content.models import User
+from app.kontres.enums import ReservationStateEnum
+from app.kontres.models.bookable_item import BookableItem
+from app.util.models import BaseModel
+
+
+class Reservation(BaseModel, BasePermissionModel):
+    read_access = [Groups.TIHLDE]
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    author = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="reservations"
+    )
+    bookable_item = models.ForeignKey(
+        BookableItem, on_delete=models.PROTECT, related_name="reservations"
+    )
+    start_time = models.DateTimeField()
+    end_time = models.DateTimeField()
+    state = models.CharField(
+        max_length=15,
+        choices=ReservationStateEnum.choices,
+        default=ReservationStateEnum.PENDING,
+    )
+    description = (models.TextField(blank=True),)
+    accepted_rules = models.BooleanField(default=True)
+
+    def __str__(self):
+        return f"{self.state} - Reservation request by {self.author.first_name} {self.author.last_name} to book {self.bookable_item.name}. Created at {self.created_at}"
+
+    @classmethod
+    def has_write_permission(cls, request):
+        return cls.has_reservation_permission(request)
```

Comment:
return check_has_access(self.write_access, request). Det skal være default at en bruker som har write access kan oppdatere og slette. Hvis det skal være egne relger for det så definerer du dette i has_object_update og destroy. classmethod leses først, og så leses de andre metodene etterpå hvis classmehod er sann